### PR TITLE
Add a visual validator for Prototype name in Prototype trait

### PR DESCRIPTION
**Pull request by BrentEaston**
Similar to the Beanshell expression validator. Highlights when you prototype name doesn't match one in the module.

### CODE REVIEW

**Comment by Cattlesquat** [comment]
Since this validator isn't optionally turn-offable, I wonder if it will create any strange new warning message cascades for modules that are presently "silently working in spite of prototypes that point at nothing"? It's just one of those "I Wonder If" moments...

**Comment by BrentEaston** [comment]
It's not a 'Validator' validator. It doesn't generate any messages and is completely inactive unless you are editing a prototype in a module.

File: vassal-app/src/main/java/VASSAL/counters/UsePrototype.java [hunk 1]
```diff
@@ -323,6 +336,86 @@ public String getState() {
     public String getType() {
       return ID + nameConfig.getValueString();
     }
+
+    private String getValueString() {
+      return nameConfig.getValueString();
+    }
+
+    private class Validator extends JLabel {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/UsePrototype.java:344]: Should this be `private static`?

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/UsePrototype.java]: Re-organized and re-coded slightly. Ready to go.

File: vassal-app/src/main/java/VASSAL/counters/UsePrototype.java [hunk 2]
```diff
@@ -323,6 +336,88 @@ public String getState() {
...
+          dirty = true;
+        }
+        else {
+          validating = true;
+          setStatus(UNKNOWN);
+          SwingUtilities.invokeLater(validationThread);
+        }
+      }
+
+      class ValidationThread implements Runnable {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/UsePrototype.java:402]: Should be private?

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/UsePrototype.java]: Actually, everything in there should be private. Updated

**Pull request merged**